### PR TITLE
feat(contrib/fvuls) Add commands to obtained CPE information of network devices by executing snmp2cpe and upload to Fvuls server

### PR DESCRIPTION
**Pull request by wadda0714**
# What did you implement:
There are two commands.

1. `future-vuls discover`

　　-  Explore hosts within the CIDR range using the ping command 
　　-  Exec snmp2cpe(https://github.com/future-architect/vuls/pull/1625) to active hosts to obtain CPE
　　      Commands running internally
    　　　　`snmp2cpe v2c {IPAddr} public  | snmp2cpe convert`
　　-  Describe the information including CPE on the found hosts in a toml-formatted file.
　　Structure of toml-formatted file
```
[server.{ip}]
ip = {IpAddr}
server_name = ""
uuid = {UUID}
cpe_uri = []
fvuls_sync = false
```
　　-  Default output file is discover_list.toml
　　-  If the file already exists, append the newly found host to the toml-formatted file. Old files are renamed with a timestamp.
　　ex.) 20230912161058_discover_list.toml
```
./future-vuls discover -h
discover hosts with CIDR range. Run snmp2cpe on active host to get CPE. Default outputFile is ./discover_list.toml

Usage:
  future-vuls discover --cidr <CIDR_RANGE> --output <OUTPUT_FILE> [flags]

Examples:
future-vuls discover --cidr 192.168.0.0/24 --output discover_list.toml

Flags:
      --cidr string           cidr range
  -h, --help                  help for discover
      --output string         output file
      --snmp-version string   snmp version v1,v2c and v3. default: v2c
```

2. `future-vuls add-cpe`
    Upload CPE information on the specified(FvulsSync is true and UUID is obtained)  hosts to Fvuls

　    
　　-  Create pseudo server to Fvuls to obtain uuid  and upload  CPE to Fvuls
　　-  Default output file is discover_list.toml
```
./future-vuls add-cpe -h
Create a pseudo server in Fvuls and register CPE. Default outputFile is ./discover_list.toml

Usage:
  future-vuls add-cpe --token <VULS_TOKEN> --output <OUTPUT_FILE> [flags]

Examples:
future-vuls add-cpe --token <VULS_TOKEN>

Flags:
  -h, --help                help for add-cpe
      --http-proxy string   proxy url
      --output string       output file
  -t, --token string        future vuls token ENV: VULS_TOKEN
```

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
```
$ make build-future-vuls

$ ./future-vuls discover --cidr 192.168.0.1/24
Discovering 192.168.0.1/24...
192.168.0.1: Execute snmp2cpe...
failed to execute snmp2cpe. err: failed to execute snmp2cpe. err: exit status 1
192.168.0.2: Execute snmp2cpe...
failed to execute snmp2cpe. err: failed to execute snmp2cpe. err: exit status 1
192.168.0.4: Execute snmp2cpe...
failed to execute snmp2cpe. err: failed to execute snmp2cpe. err: exit status 1
192.168.0.6: Execute snmp2cpe...
New network device found 192.168.0.6
wrote to discover_list.toml

$ ./future-vuls add-cpe --token fvgr-686b92af-5216-11ee-a241-0a58a9feac02
Creating 1 pseudo server...
192.168.0.6: Created FutureVuls pseudo server ce024b45-1c59-5b86-1a67-e78a40dfec01
wrote to discover_list.toml

Uploading 1 server's CPE...
192.168.0.6: Uploaded CPE cpe:2.3:h:fortinet:fortigate-50e:-:*:*:*:*:*:*:*
192.168.0.6: Uploaded CPE cpe:2.3:o:fortinet:fortios:5.4.6:*:*:*:*:*:*:*
```

discover_list.toml
```
["192.168.0.10"]
  ip = "192.168.0.10"
  server_name = "192.168.0.10"
  uuid = "e811e2b1-9463-d682-7c79-a4ab37de28cf"
  cpe_uri = ["cpe:2.3:h:fortinet:fortigate-50e:-:*:*:*:*:*:*:*", "cpe:2.3:o:fortinet:fortios:5.4.6:*:*:*:*:*:*:*"]
  fvuls_sync = true
```

# Checklist:
You don't have to satisfy all of the following.

- [ ] Write tests
- [ ] Write documentation
- [x] Check that there aren't other open pull requests for the same issue/feature
- [x] Format your source code by `make fmt`
- [x] Pass the test by `make test`
- [x] Provide verification config / commands
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES

# Reference

* https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/

### CODE REVIEW

File: contrib/future-vuls/pkg/cpe/cpe.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/future-architect/vuls/contrib/future-vuls/pkg/schema"
+	"github.com/future-architect/vuls/util"
+	"golang.org/x/exp/slices"
+)
+
+type createPseudoServerInput struct {
+	APIKey     string `json:"api_key"`
```
Comment (sadayuki-matsuno) [inline contrib/future-vuls/pkg/cpe/cpe.go:20]: use camel case

File: contrib/future-vuls/pkg/cpe/cpe.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+	if err := json.Unmarshal(t, &cpeDetail); err != nil {
+		if err.Error() == "invalid character 'A' looking for beginning of value" {
+			return "", fmt.Errorf("invalid token")
+		}
+		return "", fmt.Errorf("failed to unmarshal serverDetail. err: %v", err)
+	}
+	return cpeDetail.Server.ServerName, nil
+}
+
+func sentHTTPRequest(req *http.Request, token string, proxy string) ([]byte, error) {
```
Comment (sadayuki-matsuno) [inline contrib/future-vuls/pkg/cpe/cpe.go:298]: sendHTTPRequest

**Review by sadayuki-matsuno — CHANGES_REQUESTED** [review]

File: contrib/future-vuls/cmd/main.go [hunk 3]
```diff
@@ -100,19 +84,86 @@ func main() {
...
+			}
+			if len(snmpVersion) == 0 {
+				snmpVersion = schema.SnmpVersion
+			}
+			if snmpVersion != "v1" && snmpVersion != "v2c" && snmpVersion != "v3" {
+				return fmt.Errorf("Invalid snmpVersion")
+			}
+			if err := discover.ActiveHosts(cidr, outputFile, snmpVersion); err != nil {
+				fmt.Printf("%v", err)
+				os.Exit(1)
```
Comment (sadayuki-matsuno) [inline contrib/future-vuls/cmd/main.go:110]: What is the reason for exiting without returning err?

File: contrib/future-vuls/pkg/cpe/cpe.go (hunk 1 above)
Comment (wadda0714) [reply contrib/future-vuls/pkg/cpe/cpe.go]: コード見たらAPIKeyは必要なくなってたので削除しました！

File: contrib/future-vuls/cmd/main.go [hunk 4]
```diff
@@ -100,19 +84,86 @@ func main() {
...
+			if len(outputFile) == 0 {
+				outputFile = schema.FileName
+			}
+			url := os.Getenv("VULS_URL")
+			if len(url) == 0 {
+				url = schema.RestEndPoint
+			}
+			if err := cpe.AddServerToFvuls(token, outputFile, proxy, url); err != nil {
+				fmt.Printf("%v", err)
+				os.Exit(1)
```
Comment (sadayuki-matsuno) [inline contrib/future-vuls/cmd/main.go:136]: What is the reason for exiting without returning err?

File: contrib/future-vuls/cmd/main.go [hunk 5]
```diff
@@ -100,19 +84,86 @@ func main() {
...
+			}
+			if err := cpe.AddCpeDataToFvuls(token, outputFile, proxy, url); err != nil {
+				fmt.Printf("%v", err)
+				os.Exit(1)
```
Comment (sadayuki-matsuno) [inline contrib/future-vuls/cmd/main.go:140]: What is the reason for exiting without returning err?

**Review by sadayuki-matsuno — CHANGES_REQUESTED** [review]

**Pull request merged**